### PR TITLE
Update ruby branch to v2_6_5

### PR DESCRIPTION
**Pull request by abenbachir**
Some memory leaks fixes

### CODE REVIEW

**Comment by hestolz** [comment]
what is this—the result of updating source/ext/ruby ref and rebuilding ruby.data, or some hand-picked modifications?
if the former, shouldn't we be changing the ref as well?

**Comment by abenbachir** [comment]
> what is this—the result of updating source/ext/ruby ref and rebuilding ruby.data, or some hand-picked modifications?
> if the former, shouldn't we be changing the ref as well?

That's how we do Ruby branch update.

File: installer/datafiles/ruby.data [hunk 1]
```diff
@@ -7699,7 +7698,7 @@ ${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_mon
 ${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_dummy.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_dummy.rb; 664; root; root
 ${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_stdout.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_stdout.rb; 664; root; root
 ${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_exec.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_exec.rb; 644; root; root
-${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; 664; root; root
+${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; 644; root; root
```
Comment (hestolz) [inline installer/datafiles/ruby.data]: Why is this changed? Ruby shouldn't have any impact on fluentd plugin perms

Comment (abenbachir) [reply installer/datafiles/ruby.data]: Fluentd is part of ruby gems

Comment (abenbachir) [reply installer/datafiles/ruby.data]: I think the patch Ashley created caused the permission to change from 664 to 644

Comment (abenbachir) [reply installer/datafiles/ruby.data]: but that is fine, we have the same permission with in_exec.rb and in_syslog.rb where we added a patch for that.

```
${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_syslog.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_syslog.rb; 644; root; root
${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_exec.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/in_exec.rb; 644; root; root
${{RUBY_DEST}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; ${{RUBY_INT}}/lib/ruby/gems/2.6.0/gems/fluentd-0.12.40/lib/fluent/plugin/filter_parser.rb; 644; root; root
```
